aten/src/ATen/core/LazyTensor.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <ATen/core/TensorBody.h>
+#include <ATen/core/ivalue.h>
+#include <ATen/core/function_schema.h>
+#include <c10/core/TensorImpl.h>
+
+namespace at {
+
+struct CAFFE2_API LazyTensorImpl : public TensorImpl {
+  explicit LazyTensorImpl(TensorTypeSet ts)
+      : TensorImpl(
+            ts.add(TensorTypeId::LazyTensorId),
+            caffe2::TypeMeta::Make<float>(), // Hack to support gradients
```

Comment:
HMMMMMMMMMM

Reply:
So, like, if I pass a CUDA double tensor into LazyTensor, the LazyTensor will advertise itself as a CPU float tensor? That doesn't sound very kosher.

You are kind of stuck between a rock and a hard place here, because you kind of need dtype inference (along with shape inference) to propagate this information correctly. So we are probably just going to have to be edgy here. But it would be nicer if this problem could be very prominently documented and explained, starting with this very code spot.

Reply:
For dtypes, though, I feel like JIT's inference should be "good enough" for most users. Is this not true?

Reply:
hmm, I agree that I should advertise as CUDA double tensor. I can at least get closer with CUDA float tensor by inspecting the TensorTypeSet

I'll document it though

Reply:
Still not documented?